shm/Fruit.cpp

```diff
@@ -0,0 +1,9 @@
+#include "Fruit.hpp"
+
+Fruit& Fruit::operator--() {
+    if (daysToRot_ == 0) {
+        return *this;
+    }
+    --daysToRot_;
+    return *this;
```

Comment:
```suggestion
    daysToRot_ == 0 ? return *this : --daysToRot;
    return *this;
```

maybe?

Reply:
Are you sure it would work?

Reply:
I once tried to use something like this and I don't think tenary operator can have return inside itself

Reply:
It can't, I also tried.

Reply:
Thanks, so I leave it as it is.

Reply:
so it could look like this 

```suggestion
    daysToRot_ == 0 ?  : --daysToRot;
    return *this;
```

Seems cleaner to me